File: lib/statsailr/proc_setting/proc_two.rb
module ProcTwo
  include ProcSettingModule

  source_r_file(__dir__, File.basename(__FILE__ , ".rb") + ".R")
  validate_option("data", is_a: ["SymbolR", "String"], as: "SymbolR" , required: true)

  def setting_for_t_test( setting )
    setting.libname = ""
    setting.func_name = "sts_two_t_test"
    setting.main_arg_and_how_to_treat = [ "vars" , :read_as_strvec, :no_nil]
    setting.runtime_args = {"data" => param("data")}
    setting.store_result = true
    setting.print_opt = true
  end

  def setting_for_paired( setting )
    setting.libname = ""
    setting.func_name = "sts_two_paired"
    setting.main_arg_and_how_to_treat = [ "vars" , :read_as_strvec, :no_nil]
    setting.runtime_args = {"data" => param("data")}
    setting.store_result = true
    setting.print_opt = true
  end

  def setting_for_wilcox_test( setting )
    setting.libname = ""
    setting.func_name = "sts_two_wilcox_test"
    setting.main_arg_and_how_to_treat = [ "vars" , :read_as_strvec, :no_nil]
    setting.runtime_args = {"data" => param("data")}
    setting.store_result = true
    setting.print_opt = true
  end

end

File: lib/statsailr/proc_setting/proc_two.R
sts_two_t_test = function( data , vars , ... ){
  if( (! is.character(vars)) || length(vars) != 2 ){
    stop("vars argument requires character vector with size of 2")
  }
  x = vars[1]
  y = vars[2]
  result = t.test( data[[x]], data[[y]], ... )
  return(result)
}

sts_two_paired = function( data , vars , ... ){
  if( (! is.character(vars)) || length(vars) != 2 ){
    stop("vars argument requires character vector with size of 2")
  }
  x = vars[1]
  y = vars[2]
  result = t.test( data[[x]], data[[y]], paired = TRUE, ... )
  return(result)
}

sts_two_wilcox_test = function( data , vars , ... ){
  if( (! is.character(vars)) || length(vars) != 2 ){
    stop("vars argument requires character vector with size of 2")
  }
  x = vars[1]
  y = vars[2]
  result = wilcox.test( data[[x]], data[[y]], ... )
  return(result)
}


